api/src/main/java/org/apache/iceberg/types/PrimitiveHolder.java

```diff
@@ -33,6 +33,6 @@ class PrimitiveHolder implements Serializable {
   }
 
   Object readResolve() throws ObjectStreamException {
-    return Types.fromPrimitiveString(typeAsString);
+    return Types.fromTypeName(typeAsString);
```

Comment:
This is tested by `TestSerializableTypes`. Can you add new cases for variant (and the other new types, while you're updating it)?

Reply:
The new type tests have been added in `TestSerializableTypes`. But let me refactor a little bit.